### PR TITLE
split post display into two components

### LINKED ISSUE #327: split post display into two components
Currently we use one component (`PostDisplay`) to display posts in a list view (e.g. on the frontpage, on a channel) and also to display a post on a post detail page. For #231 I'm going to have to add a bunch of functionality to that component, and I think with that added complexity it's just going to be too much to live in one component (i.e. a bunch of editing functionality that only matters on the post detail page). So! I'm going to split the `PostDisplay` into two separate components first, before I start on the #231.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #327 

#### What's this PR do?

basically we currently have the `PostDisplay` component have two different modes, one which is used in a list-view (frontpage, channel view) and another which is used on a post detail view. To work on adding post editing I need to modify the post detail version but not the list-view version, and I think it's just too much to have it all in the same component. So, this splits it out into two different components, each of which is only responsible for one thing. They still share a lot of styling, and code which is common to the two I pulled out into helper functions in `lib/posts.js`.

#### How should this be manually tested?

All functionality should be exactly as before.

### CODE REVIEW

File: static/js/components/CompactPostDisplay.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  formatPostTitle,
+  PostVotingButtons
+} from "../lib/posts"
+
+import type { Post } from "../flow/discussionTypes"
+
+class CompactPostDisplay extends React.Component {
+  props: {
+    post: Post,
+    showChannelLink?: boolean,
```
Comment (noisecapella) [inline static/js/components/CompactPostDisplay.js]: It looks like `showChannelLink` is always present, can you make it required here?

File: static/js/components/CompactPostDisplay.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  state: {
+    upvoting: boolean
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      upvoting: false
```
Comment (noisecapella) [inline static/js/components/CompactPostDisplay.js]: Should this be part of `PostVotingButtons`?

Comment (aliceriot) [reply static/js/components/CompactPostDisplay.js]: yeah that's a good idea

**Comment by aliceriot** [comment]
@noisecapella comments addressed

**Pull request merged**